src/Hooks/UseGetAxios.jsx

```diff
@@ -0,0 +1,35 @@
+import axios from "axios";
+import { useState, useEffect } from "react";
+
+export const UseGetAxios = (api, token) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [serverData, setServerData] = useState({});
+  const [serverError, setServerError] = useState(null);
```

Comment:
you can do using on useState like this
const [isLoading, setIsLoading] = useState({
isloading:false,
serverData:[],
serverError:null
});
so you have to not call three time useState

Reply:
Yes, but I prefer making separate useState , else it will make it complex while updating state and retrieving it